--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -33,7 +33,7 @@
 {{- else if .Values.image.digest -}}
 {{- printf "%s/%s@%s" .Values.image.registry .Values.image.repository .Values.image.digest }}
 {{- else -}}
-{{- printf "%s/%s:%s" .Values.image.registry .Values.image.repository (.Values.image.tag | default .Chart.AppVersion) }}
+{{- printf "%s:%s" .Values.image.repository (.Values.image.tag | default .Chart.AppVersion) }}
 {{- end -}}
 {{- end -}}
 
@@ -486,3 +486,11 @@
 {{- define "traefik.hubTokenFilePath" }}
 {{- printf "%s/%s" (.Values.hub.tokenMountPath | trimSuffix "/") "token" -}}
 {{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
