--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -62,6 +62,6 @@
 {{- else if .Values.image.digest -}}
 {{- printf "%s/%s@%s" (include "traefik.imageRegistry" .) (include "traefik.imageRepository" .) .Values.image.digest }}
 {{- else -}}
-{{- printf "%s/%s:%s%s" (include "traefik.imageRegistry" .) (include "traefik.imageRepository" .) (.Values.image.tag | default (include "traefik.defaultTag" .)) (ternary "-hardened" "" .Values.hub.hardened) }}
+{{- printf "%s:%s%s" .Values.image.repository (.Values.image.tag | default (include "traefik.defaultTag" .)) (ternary "-hardened" "" .Values.hub.hardened) }}
 {{- end -}}
 {{- end -}}
@@ -567,3 +567,11 @@
 {{- define "traefik.hubTokenFilePath" }}
 {{- printf "%s/%s" (.Values.hub.tokenMountPath | trimSuffix "/") "token" -}}
 {{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
