--- charts-original/templates/_podtemplate.tpl
+++ charts/templates/_podtemplate.tpl
@@ -68,7 +68,7 @@
       runtimeClassName: {{ . }}
       {{- end }}
       containers:
-      - image: {{ template "traefik.image-name" . }}
+      - image: "{{ template "system_default_registry" . }}{{ template "traefik.image-name" . }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         name: {{ template "traefik.fullname" . }}
         resources:
@@ -596,7 +596,8 @@
           {{- end }}
           {{- end }}
           {{- end }}
-          {{- with .Values.providers.kubernetesIngressNGINX }}
+          {{- $kubernetesIngressNGINX := (include "providers.kubernetesIngressNGINX.values" . | fromYaml) }}
+          {{- with $kubernetesIngressNGINX }}
            {{- if .enabled }}
           - "--providers.kubernetesingressnginx"
             {{- if or .watchNamespace (and $.Values.rbac.enabled $.Values.rbac.namespaced) }}
