--- charts-original/templates/_clustermesh-apiserver-generate-certs-job-spec.tpl
+++ charts/templates/_clustermesh-apiserver-generate-certs-job-spec.tpl
@@ -5,15 +5,12 @@
     metadata:
       labels:
         k8s-app: clustermesh-apiserver-generate-certs
-        {{- with .Values.clustermesh.apiserver.podLabels }}
-        {{- toYaml . | nindent 8 }}
-        {{- end }}
     spec:
       serviceAccount: clustermesh-apiserver-generate-certs
       serviceAccountName: clustermesh-apiserver-generate-certs
       containers:
         - name: certgen
-          image: {{ .Values.certgen.image.repository }}:{{ .Values.certgen.image.tag }}
+          image: {{ template "system_default_registry" . }}{{ .Values.certgen.image.repository }}:{{ .Values.certgen.image.tag }}
           imagePullPolicy: {{ .Values.certgen.image.pullPolicy }}
           command:
             - "/usr/bin/cilium-certgen"
