--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -5,6 +5,14 @@
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end }}
+{{- end }}
+
 {{/*
 Render full image name from given values, e.g:
 ```
