--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -217,3 +217,11 @@
 {{- define "identityAllocationCRD" }}
   {{- list "crd" "doublewrite-readkvstore" "doublewrite-readcrd" | has .Values.identityAllocationMode }}
 {{- end }}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end }}
+{{- end }}
