--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -15,13 +15,9 @@
 {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- if contains $name .Release.Name }}
-{{- .Release.Name | trunc 63 | trimSuffix "-" }}
-{{- else }}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 {{- end -}}
-{{- end -}}
 
 {{/*
 Common labels
@@ -57,7 +53,7 @@
 Allow k8s-app label to be overridden
 */}}
 {{- define "coredns.k8sapplabel" -}}
-{{- default .Chart.Name .Values.k8sAppLabelOverride | trunc 63 | trimSuffix "-" -}}
+{{- coalesce .Values.k8sApp .Values.k8sAppLabelOverride .Chart.Name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
 {{/*
@@ -224,3 +220,72 @@
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Set the clusterDNS service IP
+*/}}
+{{- define "clusterDNSServerIP" -}}
+{{- if .Values.service.clusterIP }}
+    {{- .Values.service.clusterIP }}
+{{ else }}
+    {{- $dnsIPs := split "," .Values.global.clusterDNS }}
+    {{- $dnsCount := len $dnsIPs }}
+    {{- if eq $dnsCount 1 }}
+        {{- .Values.global.clusterDNS -}}
+    {{- else }}
+        {{- if gt $dnsCount 1 }}
+            {{- $dnsIPs._0 -}}
+        {{ else }}
+            {{- "10.43.0.10" }}
+        {{- end }}
+    {{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Pass the clusterDNS service IP for the nodelocal config
+*/}}
+{{- define "nodelocalUpstreamDNSServerIP" -}}
+{{- if .Values.nodelocal.ipvs }}
+{{- "" -}}
+{{ else }}
+{{- (include "clusterDNSServerIP" .) -}}
+{{- end }}
+{{- end }}
+
+{{/*
+Fill the localip flag in the nodelocal CLI
+*/}}
+{{- define "nodelocalLocalIPFlag" -}}
+{{- if .Values.nodelocal.ipvs }}
+{{- "" -}}
+{{ else }}
+{{- printf ",%s" (include "clusterDNSServerIP" .) -}}
+{{- end }}
+{{- end }}
+
+{{/*
+Fill the ipFamily correctly
+*/}}
+{{- define "ipFamilyPolicy" -}}
+{{- if .Values.service.ipFamilyPolicy }}
+    {{- .Values.service.ipFamilyPolicy }}
+{{ else }}
+    {{- $dnsIPs := split "," .Values.global.clusterDNS }}
+    {{- $dnsCount := len $dnsIPs }}
+    {{- if gt $dnsCount 1 }}
+        {{- "PreferDualStack" }}
+    {{ else }}
+        {{- "SingleStack" }}
+    {{- end }}
+{{- end }}
+{{- end }}
+    
