--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -24,7 +24,7 @@
 Allow k8s-app label to be overridden
 */}}
 {{- define "coredns.k8sapplabel" -}}
-{{- default .Chart.Name .Values.k8sAppLabelOverride | trunc 63 | trimSuffix "-" -}}
+{{- .Values.k8sApp | default .Chart.Name .Values.k8sAppLabelOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
 {{/*
@@ -191,3 +191,44 @@
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Set the clusterDNS service IP
+*/}}
+{{- define "clusterDNSServerIP" -}}
+{{- if .Values.service.clusterIP }}
+{{- .Values.service.clusterIP }}
+{{ else }}
+{{- coalesce .Values.global.clusterDNS "10.43.0.10" }}
+{{- end }}
+{{- end }}
+
+{{/*
+Pass the clusterDNS service IP for the nodelocal config
+*/}}
+{{- define "nodelocalUpstreamDNSServerIP" -}}
+{{- if .Values.nodelocal.ipvs }}
+{{- "" -}}
+{{ else }}
+{{- (include "clusterDNSServerIP" .) -}}
+{{- end }}
+{{- end }}
+
+{{/*
+Fill the localip flag in the nodelocal CLI
+*/}}
+{{- define "nodelocalLocalIPFlag" -}}
+{{- if .Values.nodelocal.ipvs }}
+{{- "" -}}
+{{ else }}
+{{- printf ",%s" (include "clusterDNSServerIP" .) -}}
+{{- end }}
+{{- end }}
