--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -70,11 +70,11 @@
 {{/*
 Get specific image
 */}}
-{{- define "ingress-nginx.image" -}}
+{{- define "ingress-nginx.repository" -}}
 {{- if .chroot -}}
-{{- printf "%s-chroot" .image -}}
+{{- printf "%s-chroot" .repository -}}
 {{- else -}}
-{{- printf "%s" .image -}}
+{{- printf "%s" .repository -}}
 {{- end }}
 {{- end -}}
 
@@ -261,3 +261,15 @@
     - name: modules
       mountPath: /modules_mount
 {{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "repository_or_registry_and_image" -}}
+{{ if .repository }}{{ .repository }}{{ else }}{{ .registry }}/{{ .image }}{{ end }}:{{ if .Values.global.prime.enabled }}{{ .Values.global.prime.tag }}{{ else }}{{ .tag }}{{ end }}{{ if .digest }}@{{ .digest }}{{ end }}
+{{- end -}}
