--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -19,7 +19,7 @@
 */}}
 
 {{- define "traefik.image-name" -}}
-{{- printf "%s/%s:%s" .Values.image.registry .Values.image.repository (.Values.image.tag | default .Chart.AppVersion) }}
+{{- printf "%s:%s" .Values.image.repository (.Values.image.tag | default .Chart.AppVersion) }}
 {{- end -}}
 {{/*
 Create a default fully qualified app name.
@@ -129,3 +129,11 @@
 {{ (split "@" (default $.Chart.AppVersion $.Values.image.tag))._0 }}
 {{- end -}}
 
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
