--- charts-original/values.yaml.tmpl
+++ charts/values.yaml.tmpl
@@ -89,8 +89,6 @@
   repository: "${CILIUM_REPO}"
   tag: "${CILIUM_VERSION}"
   pullPolicy: "${PULL_POLICY}"
-  # cilium-digest
-  digest: ${CILIUM_DIGEST}
   useDigest: ${USE_DIGESTS}
 
 # -- Affinity for cilium-agent.
@@ -645,7 +643,7 @@
 
 hubble:
   # -- Enable Hubble (true by default).
-  enabled: true
+  enabled: false
 
   # -- Buffer size of the channel Hubble uses to receive monitor events. If this
   # value is not set, the queue size is set to the default monitor queue size.
@@ -1102,7 +1100,7 @@
 ipam:
   # -- Configure IP Address Management mode.
   # ref: https://docs.cilium.io/en/stable/concepts/networking/ipam/
-  mode: "cluster-pool"
+  mode: "kubernetes"
   operator:
     # -- Deprecated in favor of ipam.operator.clusterPoolIPv4PodCIDRList.
     # IPv4 CIDR range to delegate to individual nodes for IPAM.
@@ -1332,7 +1330,7 @@
 
 # -- Configure prometheus metrics on the configured port at /metrics
 prometheus:
-  enabled: false
+  enabled: true
   port: 9962
   serviceMonitor:
     # -- Enable service monitors.
@@ -1530,14 +1528,6 @@
     override: ~
     repository: "${CILIUM_OPERATOR_BASE_REPO}"
     tag: "${CILIUM_VERSION}"
-    # operator-generic-digest
-    genericDigest: ${OPERATOR_GENERIC_DIGEST}
-    # operator-azure-digest
-    azureDigest: ${OPERATOR_AZURE_DIGEST}
-    # operator-aws-digest
-    awsDigest: ${OPERATOR_AWS_DIGEST}
-    # operator-alibabacloud-digest
-    alibabacloudDigest: ${OPERATOR_ALIBABACLOUD_DIGEST}
     useDigest: ${USE_DIGESTS}
     pullPolicy: "${PULL_POLICY}"
     suffix: "${CILIUM_OPERATOR_SUFFIX}"
@@ -1650,7 +1640,7 @@
   # -- Enable prometheus metrics for cilium-operator on the configured port at
   # /metrics
   prometheus:
-    enabled: false
+    enabled: true
     port: 9963
     serviceMonitor:
       # -- Enable service monitors.
@@ -1681,7 +1671,7 @@
 
 nodeinit:
   # -- Enable the node initialization DaemonSet
-  enabled: false
+  enabled: true
 
   # -- node-init image.
   image:
@@ -1763,8 +1753,6 @@
     override: ~
     repository: "${CILIUM_REPO}"
     tag: "${CILIUM_VERSION}"
-    # cilium-digest
-    digest: ${CILIUM_DIGEST}
     useDigest: ${USE_DIGESTS}
     pullPolicy: "${PULL_POLICY}"
 
@@ -2115,3 +2103,6 @@
   proxyPort: 0
   # -- The maximum time the DNS proxy holds an allowed DNS response before sending it along. Responses are sent as soon as the datapath is updated with the new IP information.
   proxyResponseMaxDelay: 100ms
+
+global:
+  systemDefaultRegistry: ""
